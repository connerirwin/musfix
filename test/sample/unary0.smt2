; Distinguish between unary ops
(qualif Neg ((i Int))  (- i))
(qualif Not ((b Bool)) (not b))

; Various errors
; (qualif _Neg ((b Bool)) (- b))
; (qualif _Not ((i Int)) (not i))
