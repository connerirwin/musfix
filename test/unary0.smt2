; Distinguish between unary ops
(qualif Neg     ((b bool))  (- b))
(qualif Not     ((b bool))  (not b))

; This should throw an error, invalid unary op
; (qualif Invalid ((v int))   (+ v))
