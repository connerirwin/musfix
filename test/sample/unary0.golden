
Preparing to find greatest fixpoint...

Inputs
--------
Reading from file test/sample/unary0.msmt
Qualifier "Neg" [Var IntS "i"] (Unary Neg (Var IntS "i"))
Qualifier "Not" [Var BoolS "b"] (Unary Not (Var BoolS "b"))

QMAP
--------
fromList []

Candidates
--------
[Candidate {solution = fromList [], validConstraints = fromList [], invalidConstraints = fromList [], label = "0"}]


Final candidates: 
0: [] (0 0)

